src/avtt/content-script.js

```diff
@@ -0,0 +1,580 @@
+    return result.replace(/\]\](\s*\+\s*)\[\[/g, '$1')
+}
+
+function parseDescription(request, description, {
+    bulletList = true,
+    notes = true,
+    tables = true,
+    bolded = true
+} = {}) {
+    // Trim lines
+    description = description.split('\n').map(s => s.trim()).join('\n');
+
+    if (!settings["subst-vtt"])
+        return description;
+        
+    notes && (description = formatNotesInDescription(description));
+    tables && (description = formatTableInDescription(description));
+    bulletList ? (description = formatBulletListsInDescription(description)) : (description = formatSeparateParagraphsInDescription(description));
+    bolded && (description = formatBoldedSectionInDescription(description));
+    const replaceCB = (dice, modifier) => {
+        return dice == "" ? modifier : `${dice}${modifier} (!!(${dice}${modifier}))`;
+    }
+    return subRolls(description, replaceCB);
+}
+
+function subDamageRolls(text) {
+    return subRolls(text, (dice, modifier) => {
+        const prefix = settings["auto-roll-damage"] ? "!(" : "!!(";
+        modifier = settings['critical-homebrew'] == CriticalRules.HOMEBREW_DOUBLE ? `)${modifier}` : `${modifier})`;
+        return dice == "" ? modifier : `${prefix}${dice}${modifier}`;
+    });
+}
+
+function damagesToRolls(damages, damage_types, crits, crit_types) {
+    return [
+        ...damages.map((value, index) => [damage_types[index], subDamageRolls(value)]),
+        ...(settings['critical-homebrew'] == CriticalRules.HOMEBREW_DOUBLE ? 
+            [] : 
+            crits.map((value, index) => ["Crit " + crit_types[index], subDamageRolls(value)])
+        )
+    ];
+}
+
+function template(rolls, col1 = "Name", col2 = "Roll") {
+    return `| ${col1} | ${col2} |
+| :--- | :--- |
+${rolls.map(([key, value]) => {
+    return `| ${key} | ${value} |`
+}).join('\n')}    
+
+`;
+}
+
+function formatTableInDescription(description) {
+    const matchTable =  new RegExp(/(^\n\n([^\n\r]+)\n([^\n\r]+)\n\n\n\n$)(^\n([^\n\r]+)\n([^\n\r]+)\n\n$)+/gm);
+    const matchHead =  new RegExp(/(^\n\n([^\n\r]+)\n([^\n\r]+)\n\n\n\n$)/gm);
+    const matchRow =  new RegExp(/(^\n([^\n\r]+)\n([^\n\r]+)\n\n$)/gm);
+
+    return description.replace(matchTable, (match) => {
+        let [_1, _2, firstHead, secondHead] = match.matchAll(matchHead).next().value;
+        
+        // Skip first as it is matched by the matchHead too
+        const [_, ...rows] = [...match.matchAll(matchRow)].map(([_1, _2, firstRow, secondRow]) => {
+            return `| ${firstRow} | ${secondRow} |`
+        });
+        return `
+
+| ${firstHead} | ${secondHead} |
+| :--- | :--- |
+${rows.join('\n')}
+
+`;
+    })
+}
+
+function formatNotesInDescription(description) {
+    return description.replace(/Notes: [^\n]*/g, (match) => `_${match}_`);
+}
+
+function formatBoldedSectionInDescription(description) {
+    return description.replace(/([^\n\.])*\.\&nbsp\;/g, match => `\n**${match}**`)
```

Comment:
If I remember correctly, `&nbsp;` have been replaced by a space in the descriptions in a recent version, so this wouldn't match anything I expect. Also unsure if this would work, it might match for some monster blocks, but looking for a `. ` is going to match a lot of stuff and you'll end up with all sorts of things being bolded randomly, which will annoy users.